src/components/DataTable.vue

```diff
@@ -0,0 +1,77 @@
+<template>
+  <el-table :data="tableData" height="88vh" @cell-click="handleCellClick">
+    <el-table-column
+      v-for="col in tableColumns"
+      :key="col.id"
+      :prop="col.prop"
+      :label="col.label"
+      :sortable="col.isSortable"
+      :column-key="col.prop"
+      :width="col.width"
+    />
+    <el-table-column prop="action" label="">
+      <template #default="table">
+        <div class="action-icons">
+          <i
+            class="fa-solid fa-pencil icon"
+            @click="
+              () => {
+                editHandler(table.row);
+                openAppModal();
+              }
+            "
+          ></i>
+          <i
+            class="fa-solid fa-trash icon"
+            @click="deleteHandler(table.row)"
+          ></i>
+        </div>
+      </template>
+    </el-table-column>
+  </el-table>
+  <el-pagination layout="prev, pager, next" :total="tableData.length">
+  </el-pagination>
+</template>
+
+<script>
+import { mapMutations } from 'vuex';
+import { OPEN_APP_MODAL } from '@/store/mutation-types';
+import { ElTable, ElTableColumn, ElPagination } from 'element-plus';
+
+export default {
+  name: 'DataTable',
+  components: {
+    ElTable,
+    ElTableColumn,
+    ElPagination,
+  },
+  props: {
+    tableData: {
+      type: Array,
+      required: true,
+    },
+    tableColumns: {
+      type: Array,
+      required: true,
+    },
+    editHandler: {
+      type: Function,
+      required: true,
+    },
+    deleteHandler: {
+      type: Function,
+      required: true,
+    },
+  },
```

Comment:
This is not the vue way to do things, you should emit events on clicking these icons, and react to these event appropriately in parent components